### ISSUE TITLE
iOS cannot connect after switching from WiFi to Cellular

### ISSUE THREAD

**Issue by ghost** [tailscale/tailscale#1940]
Both my iOS device and the Linux server are on 1.8 and both are on the same LAN.

While phone on WiFi, I ran an iperf3 against my server using the Tailscale IP and everything worked fine. After I disabled WiFi and switched to cellular, iperf3 throws an error saying it cannot connect to the server. This is using the iPerf iOS app.

Another test I did was run `python -m SimpleHTTPServer` on the server and then tried to access using the iOS device. The same behavior above can be replicated where I can access it on WiFi and then when switching to Cellular, I can no longer access the page.

Disconnecting and reconnecting in the Tailscale app fixes the issue.

**Comment by apenwarr** [comment]
@dawidkostek and @marekchowaniok, if either of you can include your iOS device's tailscale IP and a timestamp of when the problem occurred, we might be able to help here. Obviously this shouldn't be happening.

**Comment by cjbehm** [comment]
Having the same experience after dropping off wifi (manually or moving out of range of wifi).

Here's as much detail as I can think of
* Running two separate pihole+unbound servers on some hosted servers
* Accessible only on my tailscale network
* Overriding local DNS (not using Magic DNS)

Waking up on wifi, everything is fine. Then I take the dog out and lose the wifi connection, at which point I have to disconnect and reconnect tailscale to get name resolution working again (just times out otherwise). When I go back in and return to wifi, I do not have to reconnect.

I just did a test this morning while looking for this issue. Disabled wifi; DNS queries timed out. Reconnected tailscale and DNS queries were fine again.

Was going to give you an exact timestamp in this issue, but when I disabled wifi a 2nd time, DNS was fine. That feels a bit like something going stale only after a period of time.

iOS ip 100.95.168.74
Timestamps 2021-08-23 between 10:30am and 12:10am utc; sorry I can't be more exact with the timestamps.

**Comment by cjbehm** [comment]
Here's a more exact timestamp; 13:18 utc

**Comment by augs** [comment]
I have frequently seen this issue over the last 8 months or so, however when it occurs, ALL DNS resolution fails, I have to bounce the connection to get any internet connectivity functional.

**Comment by DentonGentry** [comment]
At the time this was filed there was no bugreport mechanism in the app. There is a newer report of similar behavior, which does have bugreports attached. I'm going to close this as a duplicate of the newer bug, which we have more ability to followup on.

Dup of https://github.com/tailscale/tailscale/issues/6075
